.github/workflows/publish-to-s3.yaml

```diff
@@ -0,0 +1,32 @@
+name: Publish to S3
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  upload:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        role-to-assume: arn:aws:iam::312560106906:role/hubverse-${{ github.event.repository.name }}-githubaction
```

Comment:
`role-to-assume` is an AWS IAM role that has permissions to write to the S3 bucket we're using

The role can only be assumed by:

* github actions
* from this repo
* in the main branch